src/Uno.UI.Composition/Composition/CompositionSpriteShape.skia.cs

```diff
@@ -75,6 +75,8 @@ private SKPaint TryCreateAndClearPaint(ref SKPaint? paint, bool isStroke)
 					paint.Shader = null;
 				}
 			}
+
+			paint.ColorFilter = Compositor.CurrentOpacityColorFilter;
```

Comment:
Not sure if it's expected, but you are not validating the `Compositor.CurrentOpacity != 1.0f` like you do in the [_SpriteVisual.skia.cs_](https://github.com/unoplatform/uno/pull/9439/files#diff-d91fc5e9d4dd61e2a1fd6dd8ed9576e14941d12a4917d29d9ba5b6712ae1a457R31)

Reply:
It is. Setting the filter to null will remove its application. `CurrentOpacityColorFilter` is null when the visual is opaque.